From d63795ad294575176909bab7fa3e29539a447fbd Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:59:33 -0400
Subject: [fs] btrfs: only fua the first superblock when writting supers

Message-id: <1407179109-13714-87-git-send-email-sandeen@redhat.com>
Patchwork-id: 87054
O-Subject: [RHEL7.1 PATCH 086/422] Btrfs: only fua the first superblock when writting supers
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit e8117c26b24098496b6011aabe84e43e0189a506
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Fri Jan 3 18:22:57 2014 +0800

    Btrfs: only fua the first superblock when writting supers

    We only intent to fua the first superblock in every device from
    comments, fix it.

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/disk-io.c b/fs/btrfs/disk-io.c
index 741363c..1f36540 100644
--- a/fs/btrfs/disk-io.c
+++ b/fs/btrfs/disk-io.c
@@ -3132,7 +3132,10 @@ static int write_dev_supers(struct btrfs_device *device,
    * we fua the first super.  The others we allow
    * to go down lazy.
    */
-  ret = btrfsic_submit_bh(WRITE_FUA, bh);
+  if (i == 0)
+   ret = btrfsic_submit_bh(WRITE_FUA, bh);
+  else
+   ret = btrfsic_submit_bh(WRITE_SYNC, bh);
   if (ret)
    errors++;
  }
-- 
1.7.1